app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3679,7 +3684,10 @@ private void onTextTracksChanged() {
         }
 
         // Normalize mismatching language strings
-        final String preferredLanguage = trackSelector.getPreferredTextLanguage();
+        final List<String> preferredLanguages =
+                trackSelector.getParameters().preferredTextLanguages;
+        final String preferredLanguage =
+                preferredLanguages.isEmpty() ? null : preferredLanguages.get(0);
```

Comment:
```suggestion
        final String preferredLanguage = trackSelector.getParameters().preferredTextLanguages
                .stream().findFirst().orElse(null);
```

Would write it like this (no additional variable required)